src/api/OpenWith.js

```diff
@@ -32,91 +34,126 @@ class OpenWith extends Base {
      * @param {string} fileId - Box file ID
      * @param {Function} successCallback - Success callback
      * @param {Function} errorCallback - Error callback
-     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
+     * @return {void}
      */
     getOpenWithIntegrations(fileId: string, successCallback: Function, errorCallback: Function) {
         this.get({
             id: fileId,
             successCallback: (openWithIntegrations) => {
-                const formattedOpenWithIntegrations = this.formatOpenWithData(openWithIntegrations);
-                this.fetchAppIntegrations(formattedOpenWithIntegrations, successCallback, errorCallback);
+                this.fetchAppIntegrations(openWithIntegrations, successCallback, errorCallback);
             },
             errorCallback
         });
     }
 
-    /**
-     * Formats data conveniently for the client
-     *
-     * @param {OpenWithIntegrations} openWithIntegrations - The available Open With integrations
-     * @return {Array<OpenWithIntegrationItem>} formatted open with integrations
-     */
-    formatOpenWithData(openWithIntegrations: OpenWithIntegrations): Array<any> {
-        // Eventually, a default integration will be returned outside of the items array,
-        // which should be inserted into this list of integrations.
-
-        const { items } = openWithIntegrations;
-        // Fix the API response by converting the app integration ID to a string
-        const integrations = items.map((item) => ({
-            ...item,
-            app_integration: {
-                id: item.app_integration.id.toString(),
-                type: 'app_integration'
-            }
-        }));
-
-        // Sort integrations by display_order
-        return integrations.sort(
-            (integrationA: OpenWithIntegrationItem, integrationB: OpenWithIntegrationItem) =>
-                integrationA.display_order - integrationB.display_order
-        );
-    }
-
     /**
      * Fetch app integrations info needed to render.
      *
      * @param {Array<OpenWithIntegrationItem>} openWithIntegrations - The available Open With integrations
      * @return {void}
      */
     fetchAppIntegrations = async (
-        openWithIntegrations: Array<OpenWithIntegrationItem>,
+        openWithIntegrations: OpenWithAPI,
         successCallback: Function,
         errorCallback: Function
     ) => {
+        const items = this.addDefaultToOpenWithItems(openWithIntegrations);
         this.appIntegrationsAPI = new AppIntegrationsAPI(this.options);
 
-        const appIntegrationPromises = openWithIntegrations.map((integrationItem: OpenWithIntegrationItem) => {
+        const appIntegrationPromises = items.map((integrationItem: Object) => {
```

Comment:
Is there a common shape to the `integrationItem` that type `Object` can be replaced with?

Reply:
yeah this should be an AppIntegrationAPIItem, good catch.